arci-speak-audio/src/lib.rs

```diff
@@ -0,0 +1,60 @@
+use arci::Speaker;
+use log::error;
+use rodio::Source;
+use std::{collections::HashMap, fs::File, io};
+
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+#[non_exhaustive]
+pub enum Error {
+    #[error("io: {:?}", .0)]
+    Io(#[from] std::io::Error),
+    #[error("rodio: {:?}", .0)]
+    Decoder(#[from] rodio::decoder::DecoderError),
+    #[error("rodio: {:?}", .0)]
+    Stream(#[from] rodio::StreamError),
+    #[error("rodio: {:?}", .0)]
+    Play(#[from] rodio::PlayError),
+    #[error("not found: {:?}", .0)]
+    HashNotFound(String),
+}
+
+pub struct AudioSpeaker {
+    message_to_file_path: HashMap<String, String>,
+}
+
+impl AudioSpeaker {
+    /// Creates a new `AudioSpeaker`.
+    pub fn new(hashmap: HashMap<String, String>) -> Self {
+        Self {
+            message_to_file_path: hashmap,
+        }
+    }
+
+    /// Similar to `Speaker::speak`, but returns an error when the command failed.
+    pub fn try_speak(&self, message: &str) -> Result<(), Error> {
+        match self.message_to_file_path.get(message) {
+            Some(path) => play_audio_file(path),
+            None => Err(Error::HashNotFound(message.to_string())),
+        }
+    }
+}
+
+impl Speaker for AudioSpeaker {
+    fn speak(&self, message: &str) {
+        if let Err(e) = self.try_speak(message) {
+            // TODO: Speaker trait seems to assume that speak method will always succeed.
+            error!("{}", e);
+        }
+    }
+}
+
+fn play_audio_file(path: &str) -> Result<(), Error> {
+    let (_stream, stream_handle) = rodio::OutputStream::try_default()?;
+    let file = File::open(path)?;
+    let source = rodio::Decoder::new(io::BufReader::new(file))?;
+    stream_handle.play_raw(source.convert_samples())?;
+
```

Comment:
Use Sink & sleep_until_end.
It's better to have sink instance inside of the struct. (I don't know actually)